PreFreeSurfer/PreFreeSurferPipeline.sh

```diff
@@ -452,253 +531,327 @@ log_Msg "POSIXLY_CORRECT="${POSIXLY_CORRECT}
+        rm -r ${T2wFolder}/T2wToT1wReg
+      fi
+
+      log_Msg "mkdir -p ${wdir}"
+      mkdir -p ${wdir}
+
+      ${RUN} ${HCPPIPEDIR_PreFS}/T2wToT1wReg.sh \
+        ${wdir} \
+        ${T1wFolder}/${T1wImage}_acpc \
+        ${T1wFolder}/${T1wImage}_acpc_brain \
+        ${T2wImagePath} \
+        ${T2wFolder}/${T2wImage}_acpc_brain \
+        ${T1wFolder}/${T1wImage}_acpc_dc \
+        ${T1wFolder}/${T1wImage}_acpc_dc_brain \
+        ${T1wFolder}/xfms/${T1wImage}_dc \
+        ${T1wFolder}/${T2wImage}_acpc_dc \
+        ${T1wFolder}/xfms/${T2wImage}_reg_dc
+
+  esac
+
+  # ------------------------------------------------------------------------------
+  #  Bias Field Correction: Calculate bias field using square root of the product
+  #  of T1w and T2w iamges.
+  # ------------------------------------------------------------------------------
+
+  if [ ! "${T2wInputImages}" = "NONE" ] ; then
+
+    log_Msg "Performing Bias Field Correction"
+    if [ ! -z ${BiasFieldSmoothingSigma} ] ; then
+      BiasFieldSmoothingSigma="--bfsigma=${BiasFieldSmoothingSigma}"
+    fi
+
+    log_Msg "mkdir -p ${T1wFolder}/BiasFieldCorrection_sqrtT1wXT1w"
+    mkdir -p ${T1wFolder}/BiasFieldCorrection_sqrtT1wXT1w
+
+    ${RUN} ${HCPPIPEDIR_PreFS}/BiasFieldCorrection_sqrtT1wXT1w.sh \
+      --workingdir=${T1wFolder}/BiasFieldCorrection_sqrtT1wXT1w \
+      --T1im=${T1wFolder}/${T1wImage}_acpc_dc \
+      --T1brain=${T1wFolder}/${T1wImage}_acpc_dc_brain \
+      --T2im=${T1wFolder}/${T2wImage}_acpc_dc \
+      --obias=${T1wFolder}/BiasField_acpc_dc \
+      --oT1im=${T1wFolder}/${T1wImage}_acpc_dc_restore \
+      --oT1brain=${T1wFolder}/${T1wImage}_acpc_dc_restore_brain \
+      --oT2im=${T1wFolder}/${T2wImage}_acpc_dc_restore \
+      --oT2brain=${T1wFolder}/${T2wImage}_acpc_dc_restore_brain \
+      ${BiasFieldSmoothingSigma}
+
+  else  # -- No T2w image
+
+    if [ ! "${T1wBiasCorrect}" = "NONE" ] ; then
+
+      log_Msg "Performing Bias Field Correction on T1w image only"
+
+      if [ ! -z ${BiasFieldSmoothingSigma} ] ; then
+        BiasFieldSmoothingSigma="--bfsigma=${BiasFieldSmoothingSigma}"
+      fi
+
+      ${RUN} ${HCPPIPEDIR_PreFS}/BiasFieldCorrection_T1wOnly.sh \
+        --workingdir=${T1wFolder}/BiasFieldCorrection_T1wOnly \
+        --T1im=${T1wFolder}/${T1wImage}_acpc_dc \
+        --T1brain=${T1wFolder}/${T1wImage}_acpc_dc_brain \
+        --obias=${T1wFolder}/BiasField_acpc_dc \
+        --oT1im=${T1wFolder}/${T1wImage}_acpc_dc_restore \
+        --oT1brain=${T1wFolder}/${T1wImage}_acpc_dc_restore_brain \
+        ${BiasFieldSmoothingSigma}
+
+    else
+
+      log_Msg "Skipping Bias Field Correction of T1w image"
+
+      cp ${T1wFolder}/${T1wImage}_acpc_dc.nii.gz ${T1wFolder}/${T1wImage}_acpc_dc_restore.nii.gz
+      cp ${T1wFolder}/${T1wImage}_acpc_dc_brain.nii.gz ${T1wFolder}/${T1wImage}_acpc_dc_restore_brain.nii.gz
+      fslmaths ${T1wFolder}/${T1wImage}_acpc_dc -div ${T1wFolder}/${T1wImage}_acpc_dc ${T1wFolder}/BiasField_acpc_dc
```

Comment:
This is big problem IMO -- the preceding 3 lines are generating files with names that imply something that they are not!  i.e., the "restore" suffix is the designation that bias field correction was applied, yet those particular "restore" files won't have had bias field correction.  Similarly, BiasField_acpc_dc won't be a bias field at all.  I appreciate that this was the easiest path foward for all the downstream scripts to still work (without necessitating major changes conditional on whether bias field correction was applied during PreFreeSurfer), but I really don't like it.  [I also acknowledge that we already use a similar hack for the situation in which GDC is not applied, in that we copy files to have a name as if GDC was applied.  But (1) I don't like that either, and (2) as practical matter, the GDC stuff is really just intermediate files (e.g., ones that we don't "release").  In contrast, the "restore" files are a fundamental pipeline output.].

Reply:
At a minimum, if there is no other viable workaround here (other than extensive changes to many downscript scripts), we need:
1) A set of very clear comments in the script explaining/capturing the implications of what we are about to do.
2) A `log_Warn` message that makes very clear that the `${T1wImage}_acpc_dc_restore*` images should be considered misnamed and ARE NOT bias field corrected, and that `${T1wFolder}/BiasField_acpc_dc` is not an estimate of the bias field.

Reply:
I have added the following:

```
  [--t1biascorrect=(YES|NO|NONE)]     Whether to run Bias Field Correction even when only T1w image 
                                      is present (YES default; optional). NOTE that if set to NO or NONE, 
                                      T1w_*_restore* images should be considered misnamed and 
                                      ARE NOT bias field corrected. 
                                      Also BiasField_acpc_dc.nii.gz IS NOT an estimate of the bias field.
```

``` bash
      log_Warn "Skipping Bias Field Correction of T1w image"
      log_Warn "As Bias Field Correction is skipped note that:"
      log_Warn "${T1wImage}_acpc_dc_restore* images should be considered misnamed and ARE NOT bias field corrected!"
      log_Warn "${T1wFolder}/BiasField_acpc_dc IS NOT an estimate of the bias field!"
```

Would that suffice for the time being as other solutions would require major coding efforts.

Reply:
I agree that we unfortunately have to stick with the naming conventions.  Really someone should not be running this code without some kind of bias correction though, so I'm not clear that should even be supported.

Reply:
Bias Field Correction now allways happen even when only T1w image is available.